interactive_engine/compiler/src/main/java/com/alibaba/graphscope/common/ir/procedure/GraphStoredProcedures.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2020 Alibaba Group Holding Limited.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.graphscope.common.ir.procedure;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+import org.apache.calcite.jdbc.JavaTypeFactoryImpl;
+import org.apache.calcite.rel.type.*;
+import org.apache.calcite.sql.type.SqlTypeName;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.yaml.snakeyaml.Yaml;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+public class GraphStoredProcedures implements StoredProcedures {
+    private static final Logger logger = LoggerFactory.getLogger(GraphStoredProcedures.class);
+    private final RelDataTypeFactory typeFactory;
+    private final Map<String, StoredProcedureMeta> storedProcedureMetaMap;
+
+    public GraphStoredProcedures(String procedureDir) throws IOException {
```

Comment:
Should we provide an abstraction for reading store procedure metadata？For now, we are using directory, but in the future, it may come from a web service.

Reply:
see `StoredProceduresReader.java`